org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/SocketServerLauncher.xtend

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2019 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+ 
+package org.eclipse.xtext.ide.server
+
+import com.google.inject.Guice
+import java.io.PrintWriter
+import java.net.InetSocketAddress
+import java.nio.channels.AsynchronousServerSocketChannel
+import java.nio.channels.Channels
+import org.eclipse.lsp4j.jsonrpc.Launcher
+import org.eclipse.lsp4j.services.LanguageClient
+import org.eclipse.xtext.ide.server.LanguageServerImpl
+import org.eclipse.xtext.ide.server.ServerModule
+import org.eclipse.xtext.util.internal.Log
+
+/**
+ * Allows to launch a language server vie a server socket.
+ * 
+ * Useful for debugging live language servers.
+ * 
+ * @author Jan K&ouml;hnlein - Initial contribution and API
+ * @since 2.18
+ */
+@Log
+class SocketServerLauncher {
+
+	static val HOST = '-host'
+	static val PORT = '-port'
+	static val TRACE = '-trace'
+	static val NO_VALIDATE = '-noValidate'
+
+	public static val DEFAULT_PORT = 5008
+	public static val DEFAULT_HOST = '0.0.0.0'
+
+	def static void main(String[] args) {
+		new SocketServerLauncher().launch(args)
+	}
+
+	def void launch(String[] args) {
+		try {
+			val injector = Guice.createInjector(serverModule)
+			val serverSocket = AsynchronousServerSocketChannel.open.bind(args.socketAddress)
+			LOG.info("Started server socket at " + args.socketAddress)
+			while (true) {
+				val socketChannel = serverSocket.accept.get
+				val in = Channels.newInputStream(socketChannel)
+				val out = Channels.newOutputStream(socketChannel)
+				val trace = args.trace
+				val validate = args.shouldValidate
+				val languageServer = injector.getInstance(LanguageServerImpl)
+				LOG.info("Starting Xtext Language Server for client " + socketChannel.remoteAddress)
+				val launcher = Launcher.createLauncher(languageServer, LanguageClient, in, out, validate, trace)
+				languageServer.connect(launcher.remoteProxy)
+				launcher.startListening
+				LOG.info("Xtext Language Server has been started.")
+			}
+		} catch (Throwable throwable) {
+			throwable.printStackTrace()
+		}
+	}
+
+	protected def getServerModule() {
+		new ServerModule
```

Comment:
i like that. unfortunately we dont have it in normal server launcher

Reply:
but does it actually help (we still call the consturcutor of the subclass, so the name need to be duplicated again

Reply:
We could improve on the `ServerLauncher` as well. 
Ah, I missed to add a return type!